app/editor/extensions/BlockMenu.tsx

```diff
@@ -69,33 +70,47 @@ export default class BlockMenuExtension extends Suggestion {
                   }
                 )
               );
-
-              const isEmptyDoc = state.doc.textContent === "";
-              if (!isEmptyDoc) {
-                decorations.push(
-                  Decoration.node(
-                    parent.pos,
-                    parent.pos + parent.node.nodeSize,
-                    {
-                      class: "placeholder",
-                      "data-empty-text": this.options.dictionary.newLineEmpty,
-                    }
-                  )
-                );
-              }
-            } else if (isSlash) {
-              decorations.push(
-                Decoration.node(parent.pos, parent.pos + parent.node.nodeSize, {
-                  class: "placeholder",
-                  "data-empty-text": `  ${this.options.dictionary.newLineWithSlash}`,
-                })
-              );
             }
 
             return DecorationSet.create(state.doc, decorations);
           },
         },
       }),
+      new PlaceholderPlugin([
+        {
+          cond: (
+            node: Node,
+            pos: number,
+            _parent: Node | null,
+            state: EditorState
+          ) => {
+            const $start = state.doc.resolve(pos + 1);
```

Comment:
Every single plugin condition is doing this calculation – can we move it outside and pass as an argument for perf?